src/firefly/java/edu/caltech/ipac/firefly/server/query/lc/IrsaLightCurveHandler.java

```diff
@@ -115,125 +98,92 @@ protected DataGroup extractTblFrom(File votableResult, RESULT_TABLES_IDX resultT
             DataGroup[] dataGroups = VoTableReader.voToDataGroups(votableResult.getAbsolutePath());
 
             return dataGroups[resultTable.ordinal()];
-        } catch (IOException e) {
-            e.printStackTrace();
+        }
+        catch (IOException e) {
+            LOG.error(e);
         }
         return null;
     }
 
     protected DataGroup getDataFromAPI(PeriodogramAPIRequest request, RESULT_TABLES_IDX resultTable) {
+
         try {
-            File apiResult = apiDownlaod(request);
+            File apiResult = apiDownload(request);
 
             return extractTblFrom(apiResult, resultTable);
 
-        } catch (IOException | FailedRequestException e) {
-            e.printStackTrace();
+        }
+        catch (IOException e) {
+             LOG.error(e);
         }
         return null;
     }
 
-    protected File apiDownlaod(PeriodogramAPIRequest request) throws IOException, FailedRequestException {
-        File apiResultTempFile = makeApiResultTempFile();
+    /**
+     * This method will get the "origin_table"'s parameter value from request.  The value can be a json string passing
+     * from the TimeSeries UI or passed from Gator.  This json strin gis converted to a TableServerRequest.
+     * The data is found and saved to a file.  The file name is lcInputTable.tbl.  After testing, the same file can be
+     * overwritten.  Thus, there is no need to have different file name.  After the file is saved, set the LC_FILE key
+     * parameter in the request to the "lcInputTable.tbl".
+     *
+     * Periodogram and peak table calculations are all based on this input table.  There is no need to handle the saving
+     * in other places.
+     *
+     * @param request
+     */
+    void saveInputToTable(PeriodogramAPIRequest request){
+        TableServerRequest sreq = QueryUtil.convertToServerRequest(request.getLcSource());
+        sreq.setPageSize(Integer.MAX_VALUE);               // ensure you're getting the full table
+        sreq.removeParam(INCL_COLUMNS);
+        try {
+            File file = File.createTempFile("/lcInputTable.tbl", ".tbl", ServerContext.getTempWorkDir());
```

Comment:
Did you test this?  Will it still work when you put a `/` in the file name?

Reply:
I copied & pasted it and forgot to remove the "/".  I tested many times and worked right.  But it should not needed.  I will remove it and push another commit.
 Thanks for spotting it.